### PR TITLE
Initial support for digital accelerometer ADXL345

### PR DESCRIPTION
**Pull request by conejoninja**
I saw it on the roadmap but couldn't find any reference for it, I needed support and added myself, but feel free to discard if work is done elsewhere.

Based the driver on this library: https://github.com/jakalada/Arduino-ADXL345

It's not 100% test-covered, because I'm getting this while trying to test a wrong range (and has no idea how to solve it): 

> not a valid range - "*errors.fundamental", should equal,  not a valid range - "*errors.errorString"

my test's code
```go
func TestADXL345DriverSetRangeError(t *testing.T) {
	d := initTestADXL345Driver()
	d.Start()
	gobottest.Assert(t, d.SetRange(17), errors.New("not a valid range"))
}
```

### CODE REVIEW

File: drivers/i2c/ADXL345_driver.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	powerCtl   adxl345PowerCtl
+	dataFormat adxl345DataFormat
+	bwRate     adxl345BwRate
+
+	x, y, z          float64
+	rawX, rawY, rawZ int16
+
+	Config
+}
+
```
Comment (trevrosen) [inline drivers/i2c/ADXL345_driver.go]: A little doc for this struct would be helpful

File: drivers/i2c/ADXL345_driver.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+type adxl345PowerCtl struct {
```
Comment (trevrosen) [inline drivers/i2c/ADXL345_driver.go]: A little doc for this struct would be helpful

File: drivers/i2c/ADXL345_driver.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+	link      uint8
+	autoSleep uint8
+	measure   uint8
+	sleep     uint8
+	wakeUp    uint8
+}
+
+type adxl345DataFormat struct {
```
Comment (trevrosen) [inline drivers/i2c/ADXL345_driver.go]: A little doc for this struct would be helpful

File: drivers/i2c/ADXL345_driver.go [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+	selfTest    uint8
+	spi         uint8
+	intInvert   uint8
+	fullRes     uint8
+	justify     uint8
+	sensorRange uint8
+}
+
+type adxl345BwRate struct {
```
Comment (trevrosen) [inline drivers/i2c/ADXL345_driver.go]: A little doc for this struct would be helpful

**Comment by trevrosen** [comment]
@deadprogram I'm good to merge but wanted to check with you on whether this meets the expectations you had in the roadmap.

**Comment by conejoninja** [comment]
The library it's based on has the most used features but lacking a few others.
Feel free to send any feedback about this or in general as I'm planning on adding a few more devices, it would be really helpful.

**Comment by deadprogram** [comment]
I think this looks very good. Any chance of getting an example for it like https://github.com/hybridgroup/gobot/blob/master/examples/firmata_mpu6050.go please?

**Comment by conejoninja** [comment]
Example added 😃

**Comment by deadprogram** [comment]
Thank you, @conejoninja now merging.

**Pull request merged**